erizo_controller/erizoController/roomController.js

```diff
@@ -239,20 +220,17 @@ exports.RoomController = (spec) => {
                                      `erizoId: ${getErizoQueueFromStreamId(streamId)}, ` +
                                      `retries: ${retries}, `,
                                      logger.objectToLog(options.metadata));
-                publishers[streamId] = undefined;
                 retries += 1;
+                callback('timeout-erizojs-retry');
+                streamManager.updateErizoIdForPublishedStream(streamId, null);
                 that.addPublisher(clientId, streamId, options, callback, retries);
                 return;
               }
               log.warn('message: addPublisher ErizoJS timeout no retry, ' +
                                  `retries: ${retries}, streamId: ${streamId}, ` +
                                  `erizoId: ${getErizoQueueFromStreamId(streamId)},`,
                                  logger.objectToLog(options.metadata));
-              const erizo = erizos.findById(publishers[streamId]);
-              if (erizo !== undefined) {
-                const index = erizo.publishers.indexOf(streamId);
-                erizo.publishers.splice(index, 1);
-              }
+              streamManager.updateErizoIdForPublishedStream(streamId, null);
```

Comment:
Maybe I'm getting this wrong, shouldn't we remove the stream from the list of publishedStreams?

Reply:
We do that in Client.js. The general rule is that roomController only updates the ErizoId for the published streams. The rest (adding/removing publishers, subscribers and update the state) should be done in the client